### ISSUE TITLE
Cann't configure 7kaa sources.

### ISSUE THREAD

**Issue by o01eg** [the3dfxdude/7kaa#83]
When I try to build latest commit I get error:
```
autoreconf-2.69: Entering directory `.'
autoreconf-2.69: running: autopoint --force
Copying file ABOUT-NLS
Copying file config.rpath
Copying file m4/codeset.m4
Copying file m4/fcntl-o.m4
Copying file m4/gettext.m4
Copying file m4/glibc2.m4
Copying file m4/glibc21.m4
Copying file m4/iconv.m4
Copying file m4/intdiv0.m4
Copying file m4/intl.m4
Copying file m4/intldir.m4
Copying file m4/intlmacosx.m4
Copying file m4/intmax.m4
Copying file m4/inttypes-pri.m4
Copying file m4/inttypes_h.m4
Copying file m4/lcmessage.m4
Copying file m4/lib-ld.m4
Copying file m4/lib-link.m4
Copying file m4/lib-prefix.m4
Copying file m4/lock.m4
Copying file m4/longlong.m4
Copying file m4/nls.m4
Copying file m4/po.m4
Copying file m4/printf-posix.m4
Copying file m4/progtest.m4
Copying file m4/size_max.m4
Copying file m4/stdint_h.m4
Copying file m4/threadlib.m4
Copying file m4/uintmax_t.m4
Copying file m4/visibility.m4
Copying file m4/wchar_t.m4
Copying file m4/wint_t.m4
Copying file m4/xsize.m4
Copying file po/Makefile.in.in
Copying file po/Makevars.template
Copying file po/Rules-quot
Copying file po/boldquot.sed
Copying file po/en@boldquot.header
Copying file po/en@quot.header
Copying file po/insert-header.sin
Copying file po/quot.sed
Copying file po/remove-potcdate.sin
autoreconf-2.69: running: aclocal --force --install -I m4
aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx.m4'
aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx_11.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx_11.m4'
aclocal-1.15: installing 'm4/intltool.m4' from '/usr/share/aclocal/intltool.m4'
aclocal-1.15: installing 'm4/pkg.m4' from '/usr/share/aclocal/pkg.m4'
aclocal-1.15: installing 'm4/sdl2.m4' from '/usr/share/aclocal/sdl2.m4'
aclocal-1.15: installing 'm4/ax_require_defined.m4' from '/usr/share/aclocal/ax_require_defined.m4'
aclocal-1.15: error: too many loops
aclocal-1.15: Please contact <bug-automake@gnu.org>.
 at /usr/share/automake-1.15/Automake/Channels.pm line 662.
	Automake::Channels::msg("automake", "", "too many loops") called at /usr/share/automake-1.15/Automake/ChannelDefs.pm line 212
	Automake::ChannelDefs::prog_error("too many loops") called at /usr/bin/aclocal-1.15 line 1188
autoreconf-2.69: aclocal failed with exit status: 255
```

Previous commit c327f6f81aab3f875a08bea6fe44222709952799 builds fine (I only need to add `IT_PROG_INTLTOOL` into `configure.ac`.

**Comment by the3dfxdude** [comment]
Works for me.

Your log of autoreconf differs from mine starting here at aclocal. Bolded lines do not appear in my log:

`autoreconf-2.69: running: aclocal --force --install -I m4
**aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx.m4'
aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx_11.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx_11.m4'
aclocal-1.15: installing 'm4/intltool.m4' from '/usr/share/aclocal/intltool.m4'**
aclocal-1.15: installing 'm4/pkg.m4' from '/usr/share/aclocal/pkg.m4'
aclocal-1.15: installing 'm4/sdl2.m4' from '/usr/share/aclocal/sdl2.m4'
**aclocal-1.15: installing 'm4/ax_require_defined.m4' from '/usr/share/aclocal/ax_require_defined.m4'
aclocal-1.15: error: too many loops
aclocal-1.15: Please contact <bug-automake@gnu.org>.
 at /usr/share/automake-1.15/Automake/Channels.pm line 662.
	Automake::Channels::msg("automake", "", "too many loops") called at /usr/share/automake-1.15/Automake/ChannelDefs.pm line 212
	Automake::ChannelDefs::prog_error("too many loops") called at /usr/bin/aclocal-1.15 line 1188
autoreconf-2.69: aclocal failed with exit status: 255`**

Why is your aclocal installing intltool.m4? The game does not use that. What versions of intltool and gettext do you have installed? How can I be sure that your build chain is not broken? Did you start from a clean copy of 7kaa?

**Comment by o01eg** [comment]
I start from clean copy.

I removed intltool part (`IT_PROG_INTLTOOL` in `configure.ac` patch and call intltoolize) but it doesn't help:

```
autoreconf-2.69: Entering directory `.'
autoreconf-2.69: running: autopoint --force
Copying file ABOUT-NLS
Copying file config.rpath
Copying file m4/codeset.m4
Copying file m4/fcntl-o.m4
Copying file m4/gettext.m4
Copying file m4/glibc2.m4
Copying file m4/glibc21.m4
Copying file m4/iconv.m4
Copying file m4/intdiv0.m4
Copying file m4/intl.m4
Copying file m4/intldir.m4
Copying file m4/intlmacosx.m4
Copying file m4/intmax.m4
Copying file m4/inttypes-pri.m4
Copying file m4/inttypes_h.m4
Copying file m4/lcmessage.m4
Copying file m4/lib-ld.m4
Copying file m4/lib-link.m4
Copying file m4/lib-prefix.m4
Copying file m4/lock.m4
Copying file m4/longlong.m4
Copying file m4/nls.m4
Copying file m4/po.m4
Copying file m4/printf-posix.m4
Copying file m4/progtest.m4
Copying file m4/size_max.m4
Copying file m4/stdint_h.m4
Copying file m4/threadlib.m4
Copying file m4/uintmax_t.m4
Copying file m4/visibility.m4
Copying file m4/wchar_t.m4
Copying file m4/wint_t.m4
Copying file m4/xsize.m4
Copying file po/Makefile.in.in
Copying file po/Makevars.template
Copying file po/Rules-quot
Copying file po/boldquot.sed
Copying file po/en@boldquot.header
Copying file po/en@quot.header
Copying file po/insert-header.sin
Copying file po/quot.sed
Copying file po/remove-potcdate.sin
autoreconf-2.69: running: aclocal --force --install -I m4
aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx.m4'
aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx_11.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx_11.m4'
aclocal-1.15: installing 'm4/pkg.m4' from '/usr/share/aclocal/pkg.m4'
aclocal-1.15: installing 'm4/sdl2.m4' from '/usr/share/aclocal/sdl2.m4'
aclocal-1.15: installing 'm4/ax_require_defined.m4' from '/usr/share/aclocal/ax_require_defined.m4'
aclocal-1.15: error: too many loops
aclocal-1.15: Please contact <bug-automake@gnu.org>.
 at /usr/share/automake-1.15/Automake/Channels.pm line 662.
	Automake::Channels::msg("automake", "", "too many loops") called at /usr/share/automake-1.15/Automake/ChannelDefs.pm line 212
	Automake::ChannelDefs::prog_error("too many loops") called at /usr/bin/aclocal-1.15 line 1188
autoreconf-2.69: aclocal failed with exit status: 255
```

**Comment by the3dfxdude** [comment]
But I have no explanation on how your change helps, or why it is needed. The commit you reference should essentially be a no-op.

In your ebuild, replace the autogen.sh command with the following commands:

> aclocal -I m4
> autoheader
> autoconf
> automake --add-missing

And run again with a completely fresh checkout.

**Comment by o01eg** [comment]
With separate commands it works:

```
 * autopoint
ac-wrapper: warning: invalid WANT_AUTOCONF 'latest'; ignoring.
Copying file ABOUT-NLS
Copying file config.rpath
Copying file m4/codeset.m4
Copying file m4/fcntl-o.m4
Copying file m4/gettext.m4
Copying file m4/glibc2.m4
Copying file m4/glibc21.m4
Copying file m4/iconv.m4
Copying file m4/intdiv0.m4
Copying file m4/intl.m4
Copying file m4/intldir.m4
Copying file m4/intlmacosx.m4
Copying file m4/intmax.m4
Copying file m4/inttypes-pri.m4
Copying file m4/inttypes_h.m4
Copying file m4/lcmessage.m4
Copying file m4/lib-ld.m4
Copying file m4/lib-link.m4
Copying file m4/lib-prefix.m4
Copying file m4/lock.m4
Copying file m4/longlong.m4
Copying file m4/nls.m4
Copying file m4/po.m4
Copying file m4/printf-posix.m4
Copying file m4/progtest.m4
Copying file m4/size_max.m4
Copying file m4/stdint_h.m4
Copying file m4/threadlib.m4
Copying file m4/uintmax_t.m4
Copying file m4/visibility.m4
Copying file m4/wchar_t.m4
Copying file m4/wint_t.m4
Copying file m4/xsize.m4
Copying file po/Makefile.in.in
Copying file po/Makevars.template
Copying file po/Rules-quot
Copying file po/boldquot.sed
Copying file po/en@boldquot.header
Copying file po/en@quot.header
Copying file po/insert-header.sin
Copying file po/quot.sed
Copying file po/remove-potcdate.sin
 * aclocal
 * QA Notice: 'aclocal' called by src_prepare: games-strategy/7kaa-9999
 * Use autotools.eclass instead of calling 'aclocal' directly.
ac-wrapper: warning: invalid WANT_AUTOCONF 'latest'; ignoring.
ac-wrapper: warning: invalid WANT_AUTOCONF 'latest'; ignoring.
 * autoheader
ac-wrapper: warning: invalid WANT_AUTOCONF 'latest'; ignoring.
 * autoconf
 * QA Notice: 'autoconf' called by src_prepare: games-strategy/7kaa-9999
 * Use autotools.eclass instead of calling 'autoconf' directly.
ac-wrapper: warning: invalid WANT_AUTOCONF 'latest'; ignoring.
 * automake
 * QA Notice: 'automake' called by src_prepare: games-strategy/7kaa-9999
 * Use autotools.eclass instead of calling 'automake' directly.
ac-wrapper: warning: invalid WANT_AUTOCONF 'latest'; ignoring.
configure.ac:96: warning: The 'AM_PROG_MKDIR_P' macro is deprecated, and its use is discouraged.
configure.ac:96: You should use the Autoconf-provided 'AC_PROG_MKDIR_P' macro instead,
configure.ac:96: and use '$(MKDIR_P)' instead of '$(mkdir_p)'in your Makefile.am files.
configure.ac:57: installing './compile'
configure.ac:82: installing './config.guess'
configure.ac:82: installing './config.sub'
configure.ac:11: installing './install-sh'
configure.ac:11: installing './missing'
src/Makefile.am: installing './depcomp'
```

But when I change command `aclocal -I m4` to `aclocal -I m4 --force --install` I get that error.

**Comment by the3dfxdude** [comment]
Do a fresh checkout. Then create the file m4/ax_require_defined.m4 shown here:
http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blob_plain;f=m4/ax_require_defined.m4

Then run using autoreconf -vif

**Comment by o01eg** [comment]
I copied `/usr/share/aclocal/ax_require_defined.m4` and autoreconf -vif successfully completed.

```
autoreconf-2.69: Entering directory `.'
autoreconf-2.69: running: autopoint --force
Copying file ABOUT-NLS
Copying file config.rpath
Copying file m4/codeset.m4
Copying file m4/fcntl-o.m4
Copying file m4/gettext.m4
Copying file m4/glibc2.m4
Copying file m4/glibc21.m4
Copying file m4/iconv.m4
Copying file m4/intdiv0.m4
Copying file m4/intl.m4
Copying file m4/intldir.m4
Copying file m4/intlmacosx.m4
Copying file m4/intmax.m4
Copying file m4/inttypes-pri.m4
Copying file m4/inttypes_h.m4
Copying file m4/lcmessage.m4
Copying file m4/lib-ld.m4
Copying file m4/lib-link.m4
Copying file m4/lib-prefix.m4
Copying file m4/lock.m4
Copying file m4/longlong.m4
Copying file m4/nls.m4
Copying file m4/po.m4
Copying file m4/printf-posix.m4
Copying file m4/progtest.m4
Copying file m4/size_max.m4
Copying file m4/stdint_h.m4
Copying file m4/threadlib.m4
Copying file m4/uintmax_t.m4
Copying file m4/visibility.m4
Copying file m4/wchar_t.m4
Copying file m4/wint_t.m4
Copying file m4/xsize.m4
Copying file po/Makefile.in.in
Copying file po/Makevars.template
Copying file po/Rules-quot
Copying file po/boldquot.sed
Copying file po/en@boldquot.header
Copying file po/en@quot.header
Copying file po/insert-header.sin
Copying file po/quot.sed
Copying file po/remove-potcdate.sin
autoreconf-2.69: running: aclocal --force --install -I m4
aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx.m4'
aclocal-1.15: overwriting 'm4/ax_cxx_compile_stdcxx_11.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx_11.m4'
aclocal-1.15: installing 'm4/pkg.m4' from '/usr/share/aclocal/pkg.m4'
aclocal-1.15: installing 'm4/sdl2.m4' from '/usr/share/aclocal/sdl2.m4'
autoreconf-2.69: configure.ac: tracing
autoreconf-2.69: configure.ac: not using Libtool
autoreconf-2.69: running: /usr/bin/autoconf-2.69 --force
autoreconf-2.69: running: /usr/bin/autoheader-2.69 --force
autoreconf-2.69: running: automake --add-missing --copy --force-missing
configure.ac:96: warning: The 'AM_PROG_MKDIR_P' macro is deprecated, and its use is discouraged.
configure.ac:96: You should use the Autoconf-provided 'AC_PROG_MKDIR_P' macro instead,
configure.ac:96: and use '$(MKDIR_P)' instead of '$(mkdir_p)'in your Makefile.am files.
configure.ac:57: installing './compile'
configure.ac:82: installing './config.guess'
configure.ac:82: installing './config.sub'
configure.ac:11: installing './install-sh'
configure.ac:11: installing './missing'
src/Makefile.am: installing './depcomp'
autoreconf-2.69: Leaving directory `.'
```

**Comment by the3dfxdude** [comment]
OK that dependency for ax_cxx_compile_stdcxx is automatically added by aclocal, as well as overwriting ax_cxx_compile_stdcxx causes it to go into a loop? I guess. At least I read that aclocal will eventually get replaced, so this silliness. Also, I don't know why autoconf-archive is even a package, because if it wasn't present, aclocal wouldn't be braindead. Although next time I'll check the m4 code more closely. Fixed in commit 0656f2e53b8efbba15cdb3644df2f8231c678913
